### PR TITLE
Term suggest

### PR DESCRIPTION
**Pull request by greg606**
Provides fuzzy search term fixing. 
When elastic search finds better phrase it is provided to the user.

In  template it can be then use like this: 

```
Did you mean <a href="{{ path(app.request.attributes.get('_route'),
            app.request.query.all|merge({'q': searchResults.suggestedTerm}))|replace({'%20': '+'})  }}">{{ searchResults.suggestedTerm }}</a> ?
```

License: AGPLv3

### CODE REVIEW

File: src/SWP/Bundle/ElasticSearchBundle/Loader/SearchResultLoader.php [hunk 1]
```diff
@@ -110,6 +110,8 @@ public function load($metaType, $withParameters = [], $withoutParameters = [], $
             }
         }
 
+        $metaCollection->suggestedTerm = $repository->getSuggestedTerm($criteria->getTerm());
```
Comment (ahilles107) [inline src/SWP/Bundle/ElasticSearchBundle/Loader/SearchResultLoader.php:113]: we need some behat tests covering this part (in behat you have elastic running, and twig template can be tested)

File: src/SWP/Bundle/ElasticSearchBundle/Repository/ArticleRepository.php [hunk 2]
```diff
@@ -138,4 +139,35 @@ public function findByCriteria(Criteria $criteria, array $extraFields = [], bool
...
+                'phrase' => ['field' => '_all']
+            ]
+        );
+
+        $query->setSuggest($suggest);
+
+        $adapter = $this->createPaginatorAdapter($query);
+        $suggest = $adapter->getSuggests();
+
+        return $suggest['phrase'][0]['options'][0]['text'] ?? '';
```
Comment (ahilles107) [inline src/SWP/Bundle/ElasticSearchBundle/Repository/ArticleRepository.php:170]: please check if keys exists before using them

File: features/twig/loaders/searchresults_loader_term_suggest.feature [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        {% else %}
+        {% if searchResults.suggestedTerm %}
+        suggested term: {{ searchResults.suggestedTerm }}
+        {% endif %}
+
+    {% endgimmelist %}
+     """
+    Then rendered template should not contain "Test Article"
+    And rendered template should contain "suggested term: article"
```
Comment (ahilles107) [inline features/twig/loaders/searchresults_loader_term_suggest.feature:52]: 👍

File: src/SWP/Bundle/ElasticSearchBundle/Repository/ArticleRepository.php (hunk 2 above)
Comment (greg606) [reply src/SWP/Bundle/ElasticSearchBundle/Repository/ArticleRepository.php:170]: this null coalescing operator `??` does that even for nested keys. it's quite handy.

**Review by takeit — APPROVED** [review]

**Pull request merged**